### ISSUE TITLE
Error: Path contains invalid characters

### ISSUE THREAD

**Issue by makspooh** [storyblok/gatsby-source-storyblok#66]
Hello guys.

I have been trying to run the project for several days and keep getting the same error:
```
Error: Path contains invalid characters: C:\Users\USER_NAME\projects\PROJECT_NAME\.cache\caches\sb-https:\a.storyblok.com\f\141501\1738x1160\92c71418af\photo-1461749280684-dccba630e2f6-1.jpeg
```

Below is my `gatsby.config.js` file:
```javascript
const path = require('path')
require('dotenv').config()

module.exports = {
  siteMetadata: {
    siteUrl: "https://SITE_URL",
    title: "TITLE",
    description: "DESCRIPTION",
    headline: "HEADLINE",
    siteLanguage: "en",
    logo: path.resolve(__dirname, "src", "images", "favicon.png"),
    url: "https://URL",
    image: path.resolve(__dirname, "src", "images", "favicon.png"),
    twitterUsername: "@TWITTER_USER_NAME",
  },
  plugins: [
    {
      resolve: "gatsby-plugin-mailchimp",
      options: {
        endpoint: "https://ENDPOINT",
        timeout: 3500, 
      },
    },
    {
      resolve: "gatsby-plugin-manifest",
      options: {
        name: "NAME",
        short_name: "SHORT_NAME",
        start_url: "/",
        icon: path.resolve(__dirname, "src", "images", "favicon.png"),
      },
    },
    "gatsby-plugin-react-helmet",
    "gatsby-plugin-use-query-params",
    "gatsby-plugin-preload-fonts",
    "gatsby-plugin-postcss",
    {
      resolve: "gatsby-source-filesystem",
      options: {
        name: "images",
        path: "./src/images/",
      },
      __key: "images",
    },
    {
      resolve: "gatsby-source-filesystem",
      options: {
        name: "pages",
        path: "./src/pages/",
      },
      __key: "pages",
    },
    {
      resolve: "gatsby-source-storyblok",
      options: {
        accessToken: process.env.STORYBLOK_API_KEY,
        version: process.env.NODE_ENV === "production" ? "published" : "draft",
        localAssets: true,
      },
    },
    "gatsby-plugin-image",
    "gatsby-plugin-sharp",
    "gatsby-transformer-sharp",
    {
      resolve: "gatsby-plugin-segment-js",
      options: {
        prodKey: "SEGMENT_PROD_WRITE_KEY",
        devKey: "SEGMENT_DEV_WRITE_KEY",
        trackPage: true,
        trackPageDelay: 50,
        delayLoad: false,
        delayLoadTime: 1000,
        manualLoad: true,
      },
    },
  ],
};
```

Next you can see the full error log:
```
> set ENABLE_GATSBY_REFRESH_ENDPOINT=true && npx gatsby develop

warn Plugin gatsby-plugin-segment-js is not compatible with your gatsby version 4.0.0 - It requires gatsby@^3.0.0 || ^2.0.0
warn Plugin gatsby-plugin-segment-js is not compatible with your gatsby version 4.0.0 - It requires gatsby@^3.0.0 || ^2.0.0
success open and validate gatsby-configs, load plugins - 1.557s
success onPreInit - 0.045s
success initialize cache - 0.201s
success copy gatsby files - 0.474s
success Compiling Gatsby Functions - 0.844s
success onPreBootstrap - 0.888s
success createSchemaCustomization - 0.018s

 ERROR

Path contains invalid characters: C:\Users\maxim\projects\PROJECT_NAME\.cache\caches\sb-https:\a.storyblok.com\f\141501\1738x1160\92c71418af\photo-1461749280684-dccba630e2f6-1.jpeg



  Error: Path contains invalid characters: C:\Users\maxim\projects\PROJECT_NAME\.cache\caches\sb-https:\a.storyblok.com\f\141501\1738x1160\92c71418af\photo-1461749280684-dccba630e2f6-1.jpeg

  - utils.js:16 checkPath
    [PROJECT_NAME]/[fs-extra]/lib/mkdirs/utils.js:16:21

  - make-dir.js:21 Object.ensureDirSync
    [PROJECT_NAME]/[fs-extra]/lib/mkdirs/make-dir.js:21:3

  - cache-lmdb.ts:42 GatsbyCacheLmdb.init
    [PROJECT_NAME]/[gatsby]/src/utils/cache-lmdb.ts:42:8

  - get-cache.ts:11 getCache
    [PROJECT_NAME]/[gatsby]/src/utils/get-cache.ts:11:45

  - gatsby-node.js:118
    [PROJECT_NAME]/[gatsby-source-storyblok]/gatsby-node.js:118:38

  - Array.forEach

  - gatsby-node.js:114 Object.exports.onCreateNode
    [PROJECT_NAME]/[gatsby-source-storyblok]/gatsby-node.js:114:18

  - api-runner-node.js:430 runAPI
    [PROJECT_NAME]/[gatsby]/src/utils/api-runner-node.js:430:22

  - api-runner-node.js:581 Promise.catch.decorateEvent.pluginName
    [PROJECT_NAME]/[gatsby]/src/utils/api-runner-node.js:581:13

  - debuggability.js:384 Promise._execute
    [PROJECT_NAME]/[bluebird]/js/release/debuggability.js:384:9

  - promise.js:518 Promise._resolveFromExecutor
    [PROJECT_NAME]/[bluebird]/js/release/promise.js:518:18

  - promise.js:103 new Promise
    [PROJECT_NAME]/[bluebird]/js/release/promise.js:103:10

  - api-runner-node.js:579
    [PROJECT_NAME]/[gatsby]/src/utils/api-runner-node.js:579:16

  - util.js:16 tryCatcher
    [PROJECT_NAME]/[bluebird]/js/release/util.js:16:23

  - reduce.js:166 Object.gotValue
    [PROJECT_NAME]/[bluebird]/js/release/reduce.js:166:18

  - reduce.js:155 Object.gotAccum
    [PROJECT_NAME]/[bluebird]/js/release/reduce.js:155:25


not finished source and transform nodes - 1.881s
```

And finally output of `gatsby info` command:
```
  System:
    OS: Windows 10 10.0.19044
    CPU: (4) x64 AMD Ryzen 3 3200U with Radeon Vega Mobile Gfx
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.10.2 - /c/Python310/python
  Browsers:
    Chrome: 98.0.4758.102
    Edge: Spartan (44.19041.1266.0), Chromium (98.0.1108.55)
  npmPackages:
    gatsby: ^4.0.0 => 4.0.0
    gatsby-cli: ^4.0.0 => 4.0.0
    gatsby-image: ^3.11.0 => 3.11.0
    gatsby-plugin-image: ^2.5.1 => 2.5.1
    gatsby-plugin-mailchimp: ^5.2.2 => 5.2.2
    gatsby-plugin-manifest: ^4.2.0 => 4.2.0
    gatsby-plugin-postcss: ^5.0.0 => 5.0.0
    gatsby-plugin-preload-fonts: ^3.0.0 => 3.0.0
    gatsby-plugin-react-helmet: ^5.5.0 => 5.5.0
    gatsby-plugin-segment-js: ^3.7.0 => 3.7.0
    gatsby-plugin-sharp: ^4.5.1 => 4.5.1
    gatsby-plugin-use-query-params: ^1.0.1 => 1.0.1
    gatsby-source-filesystem: ^4.0.0 => 4.4.0
    gatsby-source-storyblok: ^3.1.0 => 3.1.0
    gatsby-storyblok-image: ^0.3.0 => 0.3.0
    gatsby-transformer-sharp: ^4.5.0 => 4.5.0
  npmGlobalPackages:
    gatsby-cli: 4.7.0
```

Can somebody assist me with this issue?

P.S. this error is gone after deleting `gatsby-source-storyblok` plugin from `gatsby.config.js` file.

Thanks in advance!

**Comment by gracjan-op** [comment]
This issue is there because you can create `sb-https:` this name as folder name in Windows. Solution would be to rename those names to so more "proper" names for Windows systems.

**Comment by davshoward** [comment]
Same issue here - a fix would be great...

**Comment by mikaelbang** [comment]
Same here....

**Comment by gracjan-op** [comment]
The Solution would be here if in this block of code in `gatsby-node` file we replace colon (:) or entire `https:` string from `mediaDataCacheKey` by for example `mediaDataCacheKey.replace("https:", "")` . What do you think guys ? This issue is very important because it prevents windows users using this plugin so please fix it asap.

```
 if (fileNode.id) {
            fileNodeID = fileNode.id;
            await cache.set(mediaDataCacheKey, {
              fileNodeID,
              updatedAt: node.cv,
            });
          }
          ```

**Comment by maxmhoff** [comment]
Same issue here

**Comment by ArcanisCz** [comment]
Same issue here. Can we somehow make this visible to storyblok people?

**Comment by schabibi1** [comment]
Hi there! Thanks for the heads-up and we'll have a look at this issue along with a few upcoming features for this SDK.

**Comment by davshoward** [comment]
Is there any roadmap on when a fix for this will be released?

**Comment by SamHoltby** [comment]
Same issue here. This is actually preventing me from choosing Storyblok as our main CMS. Any updates @schabibi1?

**Comment by schabibi1** [comment]
Hi @davshoward and @SamHoltby There is a plan to work on in this sprint. Hopefully to bring you good news in the following days/weeks.

**Comment by YariKoen** [comment]
Hey guys, any update on this? It seems really blocking work with images within Storyblok in Gatsby.

**Comment by YariKoen** [comment]
Hello @schabibi1, sorry for the ping. I need to add products to my new affiliate shop because of some partners, but I am unable to do it since I cannot use images from Storyblok (and store them elsewhere and link URLs is overkill). The needed fix in linked PR is a one-liner, so is there anything I can do so this quick fix can be merged & deployed?

**Comment by davshoward** [comment]
> Hi @davshoward and @SamHoltby There is a plan to work on in this sprint. Hopefully to bring you good news in the following days/weeks.

Any update on this?

**Comment by YariKoen** [comment]
Hey guys, any update on this? Sorry for another ping, but I am losing money with my project and really need to make pictures work...

**Comment by davshoward** [comment]
> Hi @davshoward and @SamHoltby There is a plan to work on in this sprint. Hopefully to bring you good news in the following days/weeks.

Hey, is there any update on this? It's been a good few months since this message.

**Comment by schabibi1** [comment]
Hi, @YariKoen @davshoward 
Sorry for the long silence.
DevRels at Storyblok have been traveling intensively in many countries for conferences for these months.

We merged a fix for this issue and please update [gatsby-source-storyblok](https://github.com/storyblok/gatsby-source-storyblok) in v4.2.1 to test in your environment.
Let us know when you try it out.

[Fix/invalid char image windows #147](https://github.com/storyblok/gatsby-source-storyblok/pull/147)
